terraform-mode.el

```diff
@@ -286,6 +286,14 @@
     (kill-new url)
     (message "Copied URL: %s" url)))
 
+(defun terraform-insert-doc-in-comment ()
+  "Insert a comment containing an URL documenting the resource at point."
+  (interactive)
+  (let ((doc-url (terraform--resource-url-at-point)))
+    (unless (looking-at-p "^resource\\|^data")
+      (re-search-backward "^resource\\|^data" nil t))
+    (insert (format "# %s\n" doc-url))))
```

Comment:
I think it would be better UX if the point didn't move to the inserted comment.  You would rarely want to position the point there.  In elisp, you can prevent any point movement (or rather *restore* the original position) by wrapping any block of code with the `save-excursion` macro.

```suggestion
    (save-excursion
      (unless (looking-at-p "^resource\\|^data")
        (re-search-backward "^resource\\|^data" nil t))
      (insert (format "# %s\n" doc-url)))))
```